### ISSUE TITLE
Access policies break on MySQL strict mode

### ISSUE THREAD

**Issue by bitwolfe** [modxcms/revolution#11212]
Recently ran into an issue with access policies not being applied for resources after migrating servers. After some debugging and testing I tracked down the source of the problem.

The new server runs Percona Server 5.6 in strict configuration, which had the ONLY_FULL_GROUP_BY sql_mode set. This caused the SQL query in findPolicies() of modResource to fail, due to missing a column in the group by. Simply adding the column to the group by returns the wrong data, however.

In short: The SQL query that fetched access policies for resources does not support the sql_mode ONLY_FULL_GROUP_BY.

Forum post can be found here: http://forums.modx.com/thread/89477/resource-group-acls-being-ignored

**Comment by pbowyer** [comment]
Checking that this is not sorted in 2.X and 3.X? I've run across it today with a 2.8.3 site and wondering if a 3.X upgrade is a fix.

**Comment by Ibochkarev** [comment]
@pbowyer — a 3.x upgrade does **not** fix this. The same `GROUP BY` pattern is still in current 3.x.

### What’s broken

`modResource::findPolicy()` still runs:

```sql
SELECT Acl.target, Acl.principal, Acl.authority, Acl.policy, Policy.data
…
GROUP BY Acl.target, Acl.principal, Acl.authority, Acl.policy
```

`Policy.data` is selected but not grouped. With `sql_mode=ONLY_FULL_GROUP_BY` (MySQL 5.7+/8 default, Percona strict, etc.) the statement fails. Failed policy load → resource-group ACLs look “ignored” (as in the [forum thread](https://forums.modx.com/thread/89477/resource-group-acls-being-ignored)). Matches the original report.

bitwolfe was right that naively stuffing more columns into `GROUP BY` is the wrong first move: `Policy.data` is already determined by `Acl.policy` (`Policy.id = Acl.policy`). The intent of `GROUP BY` is only to dedupe ACL rows, not to aggregate policy JSON.

### Same footgun elsewhere

Identical (or close) `GROUP BY` without `Policy.data`:

- `modResource::findPolicy`
- `modNamespace::findPolicy`
- `Sources\modMediaSource::findPolicy`
- `modCategory::findPolicy` (xPDO `groupby` on ACL columns, still selects `Policy.data`)

`modTemplateVar::findPolicy` / `modContext::findPolicy` avoid this style (no that `GROUP BY`, or no grouping) and are the better template.

### Suggested fix

Drop the SQL `GROUP BY` and dedupe in PHP while building `$policy` (key on `target|principal|authority|policy`), same effective result as today when the query succeeds. Alternative MySQL-only: `ANY_VALUE(Policy.data)` — works on 5.7+ but is engine-specific.

Minimal sketch for Resource:

```php
$sql = "SELECT Acl.target, Acl.principal, Acl.authority, Acl.policy, Policy.data FROM {$accessTable} Acl "
    . "LEFT JOIN {$policyTable} Policy ON Policy.id = Acl.policy "
    . "JOIN {$resourceGroupTable} ResourceGroup ON Acl.principal_class = {$this->xpdo->quote(modUserGroup::class)} "
    . "AND (Acl.context_key = :context OR Acl.context_key IS NULL OR Acl.context_key = '') "
    . "AND ResourceGroup.document = :resource "
    . "AND ResourceGroup.document_group = Acl.target";
// no GROUP BY — dedupe below if needed
$seen = [];
while ($row = $query->stmt->fetch(PDO::FETCH_ASSOC)) {
    $key = $row['target'] . ':' . $row['principal'] . ':' . $row['authority'] . ':' . $row['policy'];
    if (isset($seen[$key])) {
        continue;
    }
    $seen[$key] = true;
    $policy[modAccessResourceGroup::class][$row['target']][] = [
        'principal' => $row['principal'],
        'authority' => $row['authority'],
        'policy' => $row['data'] ? $this->xpdo->fromJSON($row['data'], true) : [],
    ];
}
```

Apply the same pattern to Namespace / MediaSource / Category `findPolicy`. Worth a CI/smoke check with MySQL 8 default `sql_mode`.

Happy to PR the four `findPolicy` call sites if maintainers want that.
